File: solutions/javascript/armstrong-numbers/1/armstrong-numbers.js
//
// This is only a SKELETON file for the 'Armstrong Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isArmstrongNumber = (num) => {
  const digits = String(num).split('').map(Number);
  const power = digits.length;
  const sum = digits.reduce((acc, d) => acc + BigInt(d) ** BigInt(power), 0n);
  return sum === BigInt(num);
};

File: solutions/python/armstrong-numbers/2/armstrong_numbers.py
def is_armstrong_number(number: int) -> bool:
    """
    Check if a number is an Armstrong number.
    
    Args:
        number: The number to check.
    
    Returns:
        bool: True if the number is an Armstrong number, False otherwise.
    """
    digits = [int(d) for d in str(number)]
    num_digits = len(digits)
    armstrong_sum = sum(d ** num_digits for d in digits)
    return armstrong_sum == number